src/arche/tools/bitbucket.py

```diff
@@ -23,7 +23,7 @@ def prepare_request(url: str) -> urllib.request.Request:
 def convert_to_api_url(url: str, netloc: str, api_netloc: str) -> str:
     """Support both regular and raw URLs"""
     try:
-        user, repo, path = re.search(
+        user, repo, path = re.search(  # type: ignore
```

Comment:
Why?

Reply:
the return of `re.search` is `Optional[Match[str]]`, so mypy complains about this: `src/arche/tools/bitbucket.py:26: error: Item "None" of "Optional[Match[str]]" has no attribute "groups"`

Reply:
```def convert_to_api_url(url: str, netloc: str, api_netloc: str) -> str:
    """Support both regular and raw URLs"""
    match = re.search(f"https://{netloc}/(.*?)/(.*?)/(?:raw|src)/(.*)", url)
    if match:
        user, repo, path = match.groups()
    else:
        raise ValueError("Not a valid bitbucket URL: {url}")
    return f"https://{api_netloc}/2.0/repositories/{user}/{repo}/src/{path}"
```